cypress/integration/specialFeatures/preRollAds/testsForCanonicalOnly.js

```diff
@@ -0,0 +1,72 @@
+import { hasMedia } from '../../pages/mediaAssetPage/helpers';
+
+// const R = require('ramda');
+// const requestAdsToggle = require('./requestAdsToggle');
+
+const MIN_VIDEO_DURATION_FOR_PREROLLS = 30; // seconds
+
+const getDuration = jsonData => jsonData.promo.media.versions[0].duration;
+
+const shouldAllowAdsForAssetType = jsonData =>
+  !jsonData.metadata.blockTypes.includes('legacyMedia');
+
+// const isLiveStream = jsonData =>
+//   jsonData.;
+
+const shouldAllowAdsForVideoDuration = jsonData => {
+  if (jsonData.promo.media.live) {
+    return true;
+  }
+  return getDuration(jsonData) >= MIN_VIDEO_DURATION_FOR_PREROLLS;
+};
+
+const getAdSetting = jsonData => {
+  if (jsonData.promo.media.live) {
+    return jsonData.metadata.options.allowAdvertising;
+  }
+  return jsonData.promo.media.advertising;
+};
+
+export default ({ serviceName, pageType, path }) => {
+  describe(`Canonical tests for PreRoll ads for ${serviceName} ${pageType} ${path}`, () => {
+    it('should load pre-roll ad plugin', () => {
+      cy.request(`${path}.json`).then(({ body: jsonData }) => {
+        if (hasMedia(jsonData)) {
+          cy.log('has media');
```

Comment:
We can probably remove these logs, and just keep the logs within the else

Reply:
Removed, was just using them to help me check things